arch: riscv: always set CONFIG_ARCH to 'riscv'

gmarull opened the pull request:
Note that the _dual_ `ARCH` in RISC-V is causing issues in HWMv2, hence this cleanup. Also, it didn't make much sense, since we already have `CONFIG_64BIT`.

gmarull committed:
arch: riscv: always use 'riscv' for CONFIG_ARCH

Because the riscv32/riscv64 is redundant, one can get the same
information combining CONFIG_ARCH + CONFIG_64BIT.

Signed-off-by: Gerard Marull-Paretas <gerard@teslabs.com>

gmarull committed:
twister: drop riscv32/64 support

riscv32/64 are no longer supported, as both are the same arch: riscv.

Signed-off-by: Gerard Marull-Paretas <gerard@teslabs.com>

gmarull committed:
twister: s/riscv(32|64)/riscv

Only riscv is supported now, any 32/64-bit requirements need to use
CONFIG_64BIT now.

Signed-off-by: Gerard Marull-Paretas <gerard@teslabs.com>

gmarull commented:
note: CI issue unrelated, will rebase once fixed.

carlocaione commented:
mmmh, you sure about this? We did exactly the opposite with ARM because we had several problems keep the two together and now we are going in the opposite direction with RISCV?

npitre commented:
ARM32 and ARM64 are arguably two different architectures.
Differences between RISCV32 and RISCV64 are much much smaller.

dleach02 commented:
and, of course, CI failure in LPC, which has nothing to do with risc plaforms, is being invoked in twister... sigh

gmarull commented:
> and, of course, CI failure in LPC, which has nothing to do with risc plaforms, is being invoked in twister... sigh

We need TwisterGPT

nordicjm reviewed (changes requested):
This is actually not working, don't know why CI isn't picking it up:
```
Failed processing /tmp/aa/zephyr/tests/kernel/fpu_sharing/generic   0, error:    0
Process Process-8:
Traceback (most recent call last):
  File "/usr/lib/python3.11/multiprocessing/process.py", line 314, in _bootstrap
    self.run()
  File "/usr/lib/python3.11/multiprocessing/process.py", line 108, in run
    self._target(*self._args, **self._kwargs)
  File "/tmp/aa/zephyr/scripts/pylib/twister/twisterlib/runner.py", line 1289, in pipeline_mgr
    pb.process(pipeline, done_queue, task, lock, results)
  File "/tmp/aa/zephyr/scripts/pylib/twister/twisterlib/runner.py", line 596, in process
    ret = self.cmake(filter_stages=self.instance.filter_stages)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/tmp/aa/zephyr/scripts/pylib/twister/twisterlib/runner.py", line 1052, in cmake
    return self.run_cmake(args,filter_stages)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/tmp/aa/zephyr/scripts/pylib/twister/twisterlib/runner.py", line 401, in run_cmake
    filter_results = self.parse_generated(filter_stages)
                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/tmp/aa/zephyr/scripts/pylib/twister/twisterlib/runner.py", line 519, in parse_generated
    raise se
  File "/tmp/aa/zephyr/scripts/pylib/twister/twisterlib/runner.py", line 514, in parse_generated
    ret = expr_parser.parse(self.testsuite.filter, filter_data, edt)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/tmp/aa/zephyr/scripts/pylib/twister/expr_parser.py", line 313, in parse
    ast = parser.parse(expr_text)
          ^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/ply/yacc.py", line 333, in parse
    return self.parseopt_notrack(input, lexer, debug, tracking, tokenfunc)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/ply/yacc.py", line 1201, in parseopt_notrack
    tok = call_errorfunc(self.errorfunc, errtoken, self)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/ply/yacc.py", line 192, in call_errorfunc
    r = errorfunc(token)
        ^^^^^^^^^^^^^^^^
  File "/tmp/aa/zephyr/scripts/pylib/twister/expr_parser.py", line 179, in p_error
    raise SyntaxError("Unexpected token '%s'" % p.value)
SyntaxError: Unexpected token 'CONFIG_64BIT'
```

nordicjm commented on tests/kernel/fpu_sharing/generic/testcase.yaml:
```diff
@@ -30,8 +30,8 @@ tests:
     extra_args: PI_NUM_ITERATIONS=500
-    filter: CONFIG_CPU_HAS_FPU
-    arch_allow: riscv32
+    filter: CONFIG_CPU_HAS_FPU && !CONFIG_64BIT
```
```suggestion
    filter: CONFIG_CPU_HAS_FPU and not CONFIG_64BIT
```

gmarull replied:
thanks, good catch

nordicjm commented on tests/kernel/fpu_sharing/generic/testcase.yaml:
```diff
@@ -41,8 +41,8 @@ tests:
       - CONFIG_MAIN_STACK_SIZE=2048
-    filter: CONFIG_CPU_HAS_FPU
-    arch_allow: riscv64
+    filter: CONFIG_CPU_HAS_FPU && CONFIG_64BIT
```
```suggestion
    filter: CONFIG_CPU_HAS_FPU and CONFIG_64BIT
```

gmarull committed:
twister: add support for riscv arch

So that we can stop using the artificial riscv32/64 dual naming. This
patch temporarily allows using both, riscv or riscv32/64.

Signed-off-by: Gerard Marull-Paretas <gerard@teslabs.com>

gmarull commented:
last push: fixed some syntax errors, also updated patch set to be bisectable.

fabiobaltieri commented:
@carlescufi we settle for 3.7 for this right?
